librz/core/cdebug.c

```diff
@@ -19,9 +20,222 @@ RZ_API bool rz_core_debug_step_one(RzCore *core, int times) {
 	}
 	return true;
 }
+
+RZ_IPI void rz_core_regs_to_flags(RzCore *core) {
+	RzList *l = rz_reg_get_list(core->dbg->reg, RZ_REG_TYPE_GPR);
+	RzListIter *iter;
+	RzRegItem *reg;
+	rz_list_foreach (l, iter, reg) {
+		ut64 regval = rz_reg_get_value(core->dbg->reg, reg);
+		rz_flag_set(core->flags, reg->name, regval, reg->size);
+	}
+	rz_list_free(l);
+}
+
+RZ_IPI bool rz_core_debug_reg_list(RzDebug *dbg, int type, int size, PJ *pj, int rad, const char *use_color) {
+	int delta, cols, n = 0;
+	const char *fmt, *fmt2, *kwhites;
+	RzPrint *pr = NULL;
+	int colwidth = 20;
+	RzListIter *iter;
+	RzRegItem *item;
+	RzList *head;
+	ut64 diff;
+	char strvalue[256];
+	bool isJson = (rad == 'j' || rad == 'J');
+	if (!dbg || !dbg->reg) {
+		return false;
+	}
+	if (dbg->corebind.core) {
+		pr = ((RzCore *)dbg->corebind.core)->print;
+	}
+	if (size != 0 && !(dbg->reg->bits & size)) {
+		// TODO: verify if 32bit exists, otherwise use 64 or 8?
+		size = 32;
+	}
+	if (dbg->bits & RZ_SYS_BITS_64) {
+		//fmt = "%s = 0x%08"PFMT64x"%s";
+		fmt = "%s = %s%s";
+		fmt2 = "%s%7s%s %s%s";
+		kwhites = "         ";
+		colwidth = dbg->regcols ? 20 : 25;
+		cols = 3;
+	} else {
+		//fmt = "%s = 0x%08"PFMT64x"%s";
+		fmt = "%s = %s%s";
+		fmt2 = "%s%7s%s %s%s";
+		kwhites = "    ";
+		colwidth = 20;
+		cols = 4;
+	}
+	if (dbg->regcols) {
+		cols = dbg->regcols;
+	}
+	if (isJson) {
+		pj_o(pj);
+	}
+	// with the new field "arena" into reg items why need
+	// to get all arenas.
+
+	int itmidx = -1;
+	dbg->creg = NULL;
+	head = rz_reg_get_list(dbg->reg, type);
+	if (!head) {
+		return false;
+	}
+	rz_list_foreach (head, iter, item) {
+		ut64 value;
+		utX valueBig;
+		if (type != -1) {
+			if (type != item->type && RZ_REG_TYPE_FLG != item->type) {
+				continue;
+			}
+			if (size != 0 && size != item->size) {
+				continue;
+			}
+		}
+		// Is this register being asked?
+		if (dbg->q_regs) {
+			if (!rz_list_empty(dbg->q_regs)) {
+				RzListIter *iterreg;
+				RzList *q_reg = dbg->q_regs;
+				char *q_name;
+				bool found = false;
+				rz_list_foreach (q_reg, iterreg, q_name) {
+					if (!strcmp(item->name, q_name)) {
+						found = true;
+						break;
+					}
+				}
+				if (!found) {
+					continue;
+				}
+				rz_list_delete(q_reg, iterreg);
+			} else {
+				// List is empty, all requested regs were taken, no need to go further
+				goto beach;
+			}
+		}
+		int regSize = item->size;
+		if (regSize < 80) {
+			value = rz_reg_get_value(dbg->reg, item);
+			rz_reg_arena_swap(dbg->reg, false);
+			diff = rz_reg_get_value(dbg->reg, item);
+			rz_reg_arena_swap(dbg->reg, false);
+			delta = value - diff;
+			if (isJson) {
+				pj_kn(pj, item->name, value);
+			} else {
+				if (pr && pr->wide_offsets && dbg->bits & RZ_SYS_BITS_64) {
+					snprintf(strvalue, sizeof(strvalue), "0x%016" PFMT64x, value);
+				} else {
+					snprintf(strvalue, sizeof(strvalue), "0x%08" PFMT64x, value);
+				}
+			}
+		} else {
+			value = rz_reg_get_value_big(dbg->reg, item, &valueBig);
```

Comment:
value is never used after this assignment.

Reply:
this function is all copy-pasted code moved from rz_debug_reg_List :(

Reply:
didn't even look at it so far

Reply:
i guess you can just delete the line.